src/Components/HeaderMain.jsx

```diff
@@ -1,90 +1,161 @@
-import React from 'react';
-import HeaderCategory from './HeaderCategory'
-import { Menu, Header, Select, Container } from 'semantic-ui-react'
-import { Link } from 'react-router-dom'
-
-const countryOptions = [
-  {
-    key: "Sweden",
-    text: "Sweden",
-    value: "Sweden",
-  },
-]
-
-const cityOptions = [
-  {
-    key: "Stockholm",
-    text: "Stockholm",
-    value: "Stockholm",
-  },
-]
-
-const mainLabels = [
-  {
-    name: 'Write An Article',
-    link: '/write-article',
-    id: 'write_article'
-  }, {
-    name: 'Review Articles',
-    link: '/review-articles',
-    id: 'review_articles'
+import React, { Component } from 'react';
+import HeaderCategory from './HeaderCategory';
+import { Menu, Header, Select, Container, Divider, Segment } from 'semantic-ui-react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { signOutUser } from '../reduxTokenAuthConfig';
+
+
+class HeaderMain extends Component {
```

Comment:
Why are you using a class here?

Reply:
cause i'm passing in state from the redux store

Reply:
You can mapStateTo a function as well

Reply:
tried now again to turn it to a functional component and the `render` part goes red. and i guess i need to have a render part since i'm using if statements?

Reply:
Functional component does not have render methods, everything you have there you can write above the return statement. Functions also need to be declared.
I am not saying you have to change it here, it does not effect yet your application, just pointing out that if you are not using state or lifecycle methods (both of which you could substitute with hooks also if you want) there is no reason to use a class.
React promises performance improvements for functional components in the future, but still until than in functions you have to write less code (you don't have to use the this keyword)

Reply:
Tried with no luck. But will look at it tomorrow morning before we merge it.
Thanks for the info again :)